optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorEntityEnhancer.java

```diff
@@ -496,6 +507,25 @@ public static String generateKieRuntimeBuilder(ClassOutput classOutput, ScoreDir
         return recorderContext.newInstance(DROOLS_INITIALIZER_CLASS_NAME);
     }
 
+    private static class OptaPlannerDroolsReactiveClassVisitor extends ClassVisitor {
+        final Class<?> clazz;
+
+        public OptaPlannerDroolsReactiveClassVisitor(Class<?> clazz, ClassVisitor outputClassVisitor) {
+            super(Gizmo.ASM_API_VERSION, outputClassVisitor);
+            this.clazz = clazz;
+        }
+
+        @Override
+        public void visitEnd() {
+            super.visitEnd();
+
+            if (clazz.getAnnotation(ClassReactive.class) == null && clazz.getAnnotation(PropertyReactive.class) == null) {
```

Comment:
For the record, there are three possible values for property reactivity in Drools:

1. DISABLED ("not possible")
2. ALLOWED ("possible")
3. ENABLED ("actually enabled")

CS-D is always 1 - it's a green field project, so we have the power the disable it, and we do.
DRL is always 2, for backwards compatibility reasons.

We should not change this behavior in Quarkus.

Reply:
It is effectively the same behavior. If a class is annotated with `@ClassReactive` or `@PropertyReactive`, nothing happens. If it is not annotated with either of those annotations, it is annotated with `@ClassReactive`. This means types are Class Reactive unless is it annotated with `@PropertyReactive`, in which case they are property reactive. Drools doesn't allow setting the propertyReactive property yet in Quarkus (see https://issues.redhat.com/browse/KOGITO-5101 )